File: days_mobile/lib/models/Assessment.dart
class Assessment {
  int id;
  int classId;
  int subjectId;
  String contents;
  String assessmentDate;

  Assessment(
      {this.id,
      this.classId,
      this.subjectId,
      this.contents,
      this.assessmentDate,});

  Assessment.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    classId = json['class_id'];
    subjectId = json['subject_id'];
    contents = json['contents'];
    assessmentDate = json['assessment_date'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['class_id'] = this.classId;
    data['subject_id'] = this.subjectId;
    data['contents'] = this.contents;
    data['assessment_date'] = this.assessmentDate;
    
    return data;
  }
}

File: days/app/Models/Assessment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Assessment extends Model
{
    use HasFactory;

    protected $fillable = [
        'class_id',
        'subject_id',
        'contents',
        'assessment_date',
    ];

    public function subject()
    {
        return $this->belongsTo(Subject::class);
    }

}
